types/handler.go

```diff
@@ -7,11 +7,19 @@ type Handler func(ctx Context, msg Msg) (*Result, error)
 // If newCtx.IsZero(), ctx is used instead.
 type AnteHandler func(ctx Context, tx Tx, simulate bool) (newCtx Context, err error)
 
-// AnteDecorator wraps the next AnteHandler to perform custom pre- and post-processing.
+// PostHandler
+type PostHandler func(ctx Context, tx Tx, res *Result, simulate, success bool) (newCtx Context, err error)
```

Comment:
I don't think we should be running the post handler if the executed message has an error

Reply:
why not?